metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -1147,23 +1147,35 @@ def fetch_projects():
                 )
             # Set parent project name
             for _project_id, project in all_project_map.items():
-                if (
-                    project.parent_id is not None
-                    and project.parent_id in all_project_map
-                ):
+                if project.parent_id is None:
+                    continue
+
+                if project.parent_id in all_project_map:
                     project.parent_name = all_project_map[project.parent_id].name
+                else:
+                    self.report.warning(
+                        "Incomplete project hierarchy",
+                        f"Project details missing for {project.parent_id}. Child projects will be ingested without reference to parent project. We need Site Administrator Explorer permissions to extract complete project hierarchy.",
```

Comment:
```suggestion
                        title = "Incomplete project hierarchy",
                        message = "Project details missing. Child projects will be ingested without reference to parent project. We need Site Administrator Explorer permissions to extract complete project hierarchy.",
                        context = project.parent_id
```
https://acryldata.notion.site/Error-reporting-in-ingestion-5343cc6ea0c84633b38070d1a409c569
Not so obvious but you would notice that its easier and less verbose on UI - as context is treated more like a list on UI whereas title+subject is treated more like a grouping mechanism. I would prefer repeating same message in the list.

Reply:
Got it! I thought grouping was based only on `title`. Good to know and thanks for clarification!